.codeql-prebuild-cpp-macOS.sh

```diff
@@ -1,8 +1,24 @@
 # install dependencies for C++ analysis
 set -e
 
+# The following error occurs without arch -arm64 on macOS-14 runner:
+# Error: Cannot install under Rosetta 2 in ARM default prefix (/opt/homebrew)!
+# To rerun under ARM use:
+#     arch -arm64 brew install ...
+
+# detect if this is arm64
+#echo "uname -m: $(uname -m)"
+#echo "arch: $(arch)"
+#echo "uname -p: $(uname -p)"
+#echo "uname -a: $(uname -a)"
+#if [[ "$(uname -m)" == "arm64" ]]; then
```

Comment:
uname can fail, the best way currently is to do:

```suggestion
#if [[ $(sysctl -n machdep.cpu.brand_string) =~ "Apple" ]]; then
```

This may solve it

Reply:
I'll keep that in mind, if I could even get `arch -arm64` to work. Currently it doesn't even if I hard code it. https://github.com/LizardByte/Sunshine/actions/runs/9055643443/job/24877061140?pr=2475#step:6:48

Which is odd, because that's the suggestion that the original error gives. https://github.com/LizardByte/Sunshine/actions/runs/9054788502/job/24875053065#step:6:56

I also don't understand how the brew install only fails if it's after the code ql init step: https://github.com/github/codeql-action/issues/2266

Reply:
oh, thats odd. 

arm64e... I guess, ARMv8.3+, maybe they are using an old version of Clang, or LLVM that don't support ARMv8.3+ and they are replacing the version available to the environment?

Interesting links:
https://developer.apple.com/documentation/security/preparing_your_app_to_work_with_pointer_authentication

https://lists.llvm.org/pipermail/llvm-dev/2019-October/136091.html

https://github.com/lelegard/arm-cpusysregs/blob/main/docs/arm64e-on-macos.md